shared/libraries/motor.h

```diff
@@ -597,4 +611,88 @@ class SteeringMotor {
   bool align_complete_;             /* if calibration is previously done, use the align_angle_                              */
 };
 
+/**
+ * @brief m4310 motor class
+ */
+class Motor4310 {
+ public:
+  /** constructor wrapper over MotorCANBase
+   *  CAN frame id for different modes:
+   *      MIT:                  actual CAN id.
+   *      position-velocity:    CAN id + 0x100.
+   *      velocity:             CAN id + 0x200.
+   *  @param can    CAN object
+   *  @param rx_id  Master id
+   *  @param tx_id  CAN id *** NOT the actual CAN id but the id configured in software ***
+   *  @param mode   0: MIT
+   *                1: position-velocity
+   *                2: velocity
+   */
+  Motor4310(bsp::CAN* can, uint16_t rx_id, uint16_t tx_id, mode_t mode);
+
+  /* implements data update callback */
+  void UpdateData(const uint8_t data[]);
+
+  /* enable m4310 */
+  void MotorEnable(Motor4310* motor);
+  /* disable m4310 */
+  void MotorDisable(Motor4310* motor);
+  /* set zero position */
+  void SetZeroPos(Motor4310* motor);
+
+  /**
+   * implements transmit output specifically for 4310
+   * @param motor m4310 motor object
+   * @param mode operation modes:
+   *                0: MIT
+   *                1: position-velocity
+   *                2: velocity
+   */
+  void TransmitOutput(control::Motor4310* motor);
+
+  /* implements data printout */
+  void PrintData();
+
+  /* set output parameters for m4310 using MIT mode */
+  void SetOutput(float position, float velocity, float kp, float kd, float torque);
+
+  /* set output parameters for m4310 using position-velocity mode */
+  void SetOutput(float position, float velocity);
+
+  /* set output parameters for m4310 using velocity mode */
+  void SetOutput(float velocity);
```

Comment:
Not everyone is familiar with these modes. Suggest adding more detailed descriptions, especially MIT mode since the only source we find about this mode is the 4310 document.

For example: 
Do we need to set PID for velocity or position-velocity mode?
MIT mode has a "torque" parameter but it's set to 0 everywhere in this PR. When do we use this value?
Are we supposed to change PID values on the fly for MIT mode?